expertsystem/topology/graph.py

```diff
@@ -164,7 +168,7 @@ def attach_edges_to_node_outgoing(
             self.edges[edge_id].originating_node_id = node_id
 
     def get_originating_node_list(
-        self, edge_ids: Sequence[int]
+        self, edge_ids: Union[Set[int], Tuple[int, ...]]
```

Comment:
But I think `Sequence[int]` is the correct type here, since the code would work with any data type that implements the Sequence interface

Reply:
Strange, if I do this, `mypy` complains:
```
expertsystem/ui/_system_control.py:579: error: Argument 1 to "get_originating_node_list" of "StateTransitionGraph" has incompatible type "Set[int]"; expected "Sequence[int]"
```
So `Set` is not a `Sequence`?

Reply:
Yeah true. Sequence implies an order. So instead it should be the unordered from of sequence, I guess `Iterable`

Reply:
:heavy_check_mark: 8e8c2cc2caabaeb2c0b914b7d5e038b5bf332a7c

Reply:
Hmm, ok we have to think about this a bit more. I feel like this has to be a `Set` (since the ids have to be unique)

Reply:
So what I'm trying to say is that we have to like modify the types of the nodes and edge of a graph in the future. Since this PR is just about adding types on the current code, this issue has to be tackled separately. (see #226)

Reply:
Yes, that's better